dynamiq/nodes/llms/base.py

```diff
@@ -199,13 +208,36 @@ def _handle_streaming_completion_response(
         full_response = self._stream_chunk_builder(chunks=chunks, messages=messages)
         return self._handle_completion_response(response=full_response, config=config, **kwargs)
 
+    def _get_response_format_and_tools(
+        self, inference_mode: InferenceMode, schema: dict[str, Any]
+    ) -> tuple[dict[str, Any] | None, dict[str, Any] | None]:
+        """Get response format and tools based on inference mode and schema_.
+
+        Args:
+            inference_mode (InferenceMode): The inference mode to use.
+            schema (Dict[str, Any]): The schema to use.
+
+        Returns:
+            tuple[Optional[Dict[str, Any]], Optional[Dict[str, Any]]]: Response format and tools.
+        """
+        response_format = None
+        tools = None
+
+        match inference_mode:
```

Comment:
It would be great to validate that `schema` is not None in case of InferenceMode.STRUCTURED_OUTPUT and InferenceMode.FUNCTION_CALLING